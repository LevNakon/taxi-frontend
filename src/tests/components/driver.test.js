import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Driver from '../../components/driver';

describe('Driver --- test render correctly component', () => {
    const initialState = {
        driverState: {
            message: '',
            success: true,
            driver: {
                experience: 0,
                condition: 'COMFORT',
            }
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<Driver store={store} />);
    });

    it('render correctly component Driver', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('check props matches with initialState', () => {
        expect(wrapper.dive().prop('success')).toEqual(initialState.driverState.success);
        expect(wrapper.dive().prop('message')).toEqual(initialState.driverState.message);
        expect(wrapper.dive().prop('driver')).toEqual(initialState.driverState.driver);
    });
});