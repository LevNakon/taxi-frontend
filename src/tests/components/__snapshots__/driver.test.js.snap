// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Driver --- test render correctly component render correctly component Driver 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(Driver)
    driver={
      Object {
        "condition": "COMFORT",
        "experience": 0,
      }
    }
    driverCreateNull={[Function]}
    driverUpdateWatcher={[Function]}
    message=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    success={true}
    userGetWatcher={[Function]}
  />
</ContextProvider>
`;
